tools/functional-tester/etcd-tester/checks.go

```diff
@@ -39,39 +41,42 @@ func newHashChecker(hrg hashAndRevGetter) Checker { return &hashChecker{hrg} }
 
 const leaseCheckerTimeout = 10 * time.Second
 
-func (hc *hashChecker) Check() (err error) {
-	plog.Printf("fetching current revisions...")
+func (hc *hashChecker) checkRevAndHashes() (err error) {
+	plog.Printf("fetching current revisions and hashes...")
 	var (
-		revs   map[string]int64
-		hashes map[string]int64
-		ok     bool
+		revs       map[string]int64
+		hashes     map[string]int64
+		sameRev    bool
+		sameHashes bool
 	)
 	// retry in case of transient failure
-	for i := 0; i < 3; i++ {
+	for i := 0; i < retries; i++ {
 		revs, hashes, err = hc.hrg.getRevisionHash()
 		if err != nil {
 			plog.Printf("#%d failed to get current revisions (%v)", i, err)
 			continue
 		}
-		if _, ok = getSameValue(revs); ok {
-			break
-		}
 
-		plog.Printf("#%d inconsistent current revisions %+v", i, revs)
+		_, sameRev = getSameValue(revs)
```

Comment:
i think we also want to detect that they are stablized. which means there is no changes for the last N seconds.